diff --git a/clang/lib/Driver/ToolChains/CommonArgs.cpp b/clang/lib/Driver/ToolChains/CommonArgs.cpp
index 774a1843fa2c..d356daf80ca5 100644
--- a/clang/lib/Driver/ToolChains/CommonArgs.cpp
+++ b/clang/lib/Driver/ToolChains/CommonArgs.cpp
@@ -1781,6 +1781,8 @@ SmallString<128> tools::getStatsFileName(const llvm::opt::ArgList &Args,
     }
 
     StringRef BaseName = llvm::sys::path::filename(Input.getBaseInput());
+    if (BaseName.empty())
+      BaseName = llvm::sys::path::filename(Output.getFilename());
     llvm::sys::path::append(StatsFile, BaseName);
     llvm::sys::path::replace_extension(StatsFile, "stats");
   } else {
